internal/testrunner/runners/system/servicedeployer/custom_agent.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package servicedeployer
+
+import (
+	_ "embed"
+	"os"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/elastic-package/internal/install"
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+//go:embed custom-agent-base-config.yml
+var customAgentYml string
+
+// CustomAgentDeployer knows how to deploy a custom elastic-agent defined via
+// a Docker Compose file.
+type CustomAgentDeployer struct {
+	cfg string
+	cd  *DockerComposeServiceDeployer
+}
+
+type deployedCustomAgent struct {
+	cfg string
+	*dockerComposeDeployedService
+}
+
+var _ DeployedService = &deployedCustomAgent{}
+
+// NewCustomAgentDeployer returns a new instance of a deployedCustomAgent.
+func NewCustomAgentDeployer(cfgPath string, ymlPaths []string, sv ServiceVariant) (*CustomAgentDeployer, error) {
```

Comment:
What's the difference between `cfgPath` and `ymlPaths`? Are both used?

Reply:
cfgPath is the custom agent options and is always mandatory. If you define other containers then ymlpaths is also used, but they are optional. `cfgPath` is used to create a temporary ymlPath for the custom agent container though.